### ISSUE TITLE
Implement basis for ship objects

### ISSUE THREAD

**Issue by jghibiki** [topoftheyear/Byte-le-Royale-2019#5]
- [x] Implement ship objects 
  - make sure they are generic so that we can subclass and attach an NPC AI to them.
- [x] Implement basic upgrade system
- [x] Think about what sorts of properties might be useful for visualization, however don't go ham until the visualizer is started.

Some of this work is already done but may need to be re-evaluated after #4 is completed.

**Depends on:** #4
